src/test/cpp/src/test_genomicsdb_api.cc

```diff
@@ -0,0 +1,239 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS 
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR 
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER 
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Test the GenomicsDB query api
+ *
+ */
+
+#include <catch2/catch.hpp>
+
+#include "genomicsdb.h"
+
+#include <iostream>
+#include <string>
+#include <utility>
+
+TEST_CASE("api get_version", "[get_version]") {
+  REQUIRE(genomicsdb_version().size() > 0);
+  REQUIRE_THAT(genomicsdb_version(), Catch::Equals(GENOMICSDB_VERSION));
+}
+
+TEST_CASE("api empty_args", "[empty_args]") {
+  std::string empty_string;
+
+  // Constructor that allows workspace/callset/vidmapping/reference genome specification
+  CHECK_THROWS_AS(new GenomicsDB(empty_string, empty_string, empty_string, empty_string), std::exception);
+  CHECK_THROWS_AS(new GenomicsDB("ws", empty_string, empty_string, empty_string), std::exception);
+  CHECK_THROWS_AS(new GenomicsDB("ws", "callset", empty_string, empty_string), std::exception);
+  CHECK_THROWS_AS(new GenomicsDB("ws", "callset", "vid", empty_string), std::exception);
+
+  // Constructor that allows json files for configuration
+  CHECK_THROWS_AS(new GenomicsDB(empty_string), std::exception);
+  CHECK_THROWS_AS(new GenomicsDB(empty_string, empty_string), std::exception);
+  CHECK_THROWS_AS(new GenomicsDB(empty_string, "loader_config.json"), std::exception);
+}
+
+static std::string ctests_input_dir(GENOMICSDB_CTESTS_DIR);
+
+static std::string workspace(ctests_input_dir+"ws");
+static std::string callset_mapping(ctests_input_dir+"callset_t0_1_2.json");
+static std::string vid_mapping(ctests_input_dir+"vid.json");
+static std::string reference_genome(ctests_input_dir+"chr1_10MB.fasta.gz");
+
+static std::string query_json(ctests_input_dir+"query.json");
+static std::string loader_json(ctests_input_dir+"loader.json");
+
+static std::string array("t0_1_2");
+
+void check_query_variants_results(GenomicsDB* gdb, const std::string& array, GenomicsDBVariants variants) {
+   REQUIRE(variants.size() == 4);
+   CHECK(variants.at(5) == nullptr);
+   auto variant = variants.next();
+   auto variant1 = variants.at(0);
+   REQUIRE(variant == variant1);
+   auto interval = gdb->get_interval(variant);
+   CHECK(interval.first == 12140);
+   CHECK(interval.second == 12294);
+   auto genomic_interval = gdb->get_genomic_interval(variant);
+   CHECK(genomic_interval.contig_name == "1");
+   CHECK(genomic_interval.interval.first == 12141);
+   CHECK(genomic_interval.interval.second == 12295);
+   auto genomic_fields = gdb->get_genomic_fields(array, variant);
```

Comment:
So I understand...can you explain why genomic_fields are empty for Variants, but not for VariantCalls? I'm sure it has to do with whatever test array was ingested/imported...but I would think the ref/alt fields for instance would apply to the Variants

Reply:
Genomic Fields at the top level Variants are common fields from the contained Variant Calls. When the number of Variant Calls is 1, this field is always empty @kgururaj may have more insight here. I can open up an Issue and fix if need be.